src/Microsoft.AspNet.Mvc.ViewFeatures/Internal/ValidationHelpers.cs

```diff
@@ -23,7 +27,7 @@ public static string GetUserErrorMessageOrDefault(ModelError modelError, ModelSt
             }
 
             var attemptedValue = entry.AttemptedValue ?? "null";
-            return Resources.FormatCommon_ValueNotValidForProperty(attemptedValue);
+            return messageProvider.InvalidValueWithUnknownModelErrorAccessor(attemptedValue);
```

Comment:
This is more of what I was talking about. This isn't tied to a specific model metadata, and so it doesn't fit into the current design.